### PR TITLE
redis cache support

### PR DESCRIPTION
**Pull request by renskiy**

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -185,11 +188,31 @@ type permission struct {
...
+	CacheTTL int `toml:"cache_ttl"`
+
+	metrics    prometheus.ObserverVec
+	ctx        context.Context
+	cacheCodec *cache.Codec
 }
 
 func (p *permission) Check(claims *jwt.StandardClaims, action Action, objectValues ...string) error {
+	if p.cacheCodec == nil {
+		return p.check(claims, action, objectValues...)
```
Comment (alexvassel) [inline auth.go]: maybe do not start app, if cache is not defined. Or maybe log this case?

Comment (renskiy) [reply auth.go]: it is not necessary to have cache on every infrastructure. For instance testing and staging do not require it.

Comment (alexvassel) [reply auth.go]: Log this, please. For understanding this fact.

Comment (renskiy) [reply auth.go]: Well, I'll do it but on the service side, not here

**Pull request merged**